src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java

```diff
@@ -74,4 +79,29 @@ public PostsListPage clickOnTrendingLinkMobile() {
   public String getSortButtonLabel() {
     return labelInSortEntryPointMobile.getText();
   }
+
+  public void scrollToBottom(WebDriver driver) {
+    ((JavascriptExecutor) driver)
+        .executeScript("window.scrollTo(0, document.body.scrollHeight)");
+  }
+
+  public int getPostsListLength() {
+    wait.forElementNotVisible(By.className(".spinner"));
```

Comment:
don't wait for spinner here, instead before method invocation call `new BasePageObject(driver).waitForLoadingOverlayToDisappear();` in test case class